unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,110 @@ ProceduralAsset asset
+
+            var bounds = GetAgentBounds(agent, agentType);
+            Vector3 absoluteSizeInWorld = new Vector3(scaleRatio.x, scaleRatio.y, scaleRatio.z);
+
+            Vector3 colliderSize = new Vector3(
+                scaleRatio.x * bounds.extents.x * 2,
+                scaleRatio.y * bounds.size.y,
+                scaleRatio.z * bounds.extents.z * 2
+            );
+            if (useAbsoluteSize) {
+                colliderSize = scaleRatio;
+            }
+
+            GameObject noneTriggeredEncapsulatingBox = new GameObject("NonTriggeredEncapsulatingBox");
+            // noneTriggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, agent.transform.position.z);
+            noneTriggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, bounds.center.z);
+
+            BoxCollider nonTriggeredBoxCollider = noneTriggeredEncapsulatingBox.AddComponent<BoxCollider>();
+            nonTriggeredBoxCollider.size = colliderSize; // Scale the box to the agent's size
+            nonTriggeredBoxCollider.enabled = true;
+
+            noneTriggeredEncapsulatingBox.transform.parent = agent.transform;
+
+            GameObject triggeredEncapsulatingBox = new GameObject("triggeredEncapsulatingBox");
+            // triggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, agent.transform.position.z);
+            triggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, bounds.center.z);
+
+            BoxCollider triggeredBoxCollider = triggeredEncapsulatingBox.AddComponent<BoxCollider>();
+            triggeredBoxCollider.size = colliderSize; // Scale the box to the agent's size
+            triggeredBoxCollider.enabled = true;
+            triggeredBoxCollider.isTrigger = true;
+
+            triggeredEncapsulatingBox.transform.parent = agent.transform;
+
+            if (useVisibleColliderBase){
+                if (useAbsoluteSize) {
+                    colliderSize = new Vector3(
+                        scaleRatio.x,
+                        0.15f,
+                        scaleRatio.z
+                    );
+                } else {
+                    colliderSize = new Vector3(
+                        scaleRatio.x * bounds.extents.x * 2,
+                        0.15f,
+                        scaleRatio.z * bounds.extents.z * 2
+                    );
+                }
+                GameObject visibleBox = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                visibleBox.name = "VisibleBox";
+                visibleBox.transform.position = new Vector3(bounds.center.x, 0.05f, bounds.center.z);
+                visibleBox.transform.localScale = colliderSize;
+                visibleBox.transform.parent = agent.transform;
+            }
+
+            this.transform.rotation = originalRotation;
+        }
+
+        public void DestroyAgentBoxCollider(ServerAction action){
```

Comment:
I would replace the `ServerAction action` parameter into the more explicit syntax we have adopted that lists the exact parameters used to avoid passing in the entire ServerAction class.

This one doesn't appear to actually need any parameters so just leaving it with no parameters should be fine.